opensquirrel/writer/writer.py

```diff
@@ -59,10 +59,12 @@ def visit_gate(self, gate: Gate) -> None:
                 gate_name = gate_name.replace("\n", "")
             self.output += f"{gate_name}\n"
             return
-        if any(not isinstance(arg, Qubit) for arg in gate.arguments):  # type: ignore[union-attr]
-            params = [arg.accept(self) for arg in gate.arguments if not isinstance(arg, Qubit)]  # type: ignore[union-attr]
+        if any(not isinstance(arg, QubitLike.__args__) for arg in gate.arguments):  # type: ignore
```

Comment:
This `QubitLike.__args__` looks ugly. Code smell maybe?

Reply:
If we take this out, python 3.9 crashes `TypeError: Subscripted generics cannot be used with class and instance checks`

Reply:
This is a bug that was fixed on after but since we support 3.9, we are forced to use `__args__`

Reply:
Fair enough, thanks.